lib/Sema/ResilienceDiagnostics.cpp

```diff
@@ -111,21 +111,6 @@ bool TypeChecker::diagnoseInlinableDeclRef(SourceLoc loc,
                               TreatUsableFromInlineAsPublic).isPublic())
     return false;
 
-  // Enum cases are handled as part of their containing enum.
-  if (isa<EnumElementDecl>(D))
-    return false;
-
-  // Protocol requirements are not versioned because there's no
-  // global entry point.
-  if (isa<ProtocolDecl>(D->getDeclContext()) &&
-      D->isProtocolRequirement())
-    return false;
-
-  // Dynamic declarations are not versioned because there's no
-  // global entry point.
-  if (D->isDynamic())
-    return false;
```

Comment:
@slavapestov I really don't know what to do about this. Fixing this is source-breaking, but 4.2 rejects `@usableFromInline` on `dynamic` declarations.

Reply:
Does @usableFromInline matter for dynamic? There’s no public entry point.

Reply:
`-swift-version 4.2` has shipped. We can't do changes there. I guess we'll have to limit it to 5 mode.

We definitely can't print referenced internal dynamic methods; they may be using internal types. Actually, that's just a hole in general without this.